development/versioning-and-changelog/migration-from-cw-0.10-to-1.0.md

```diff
@@ -125,6 +125,11 @@ The sub-messages feature can be used to get the response data or events from the
 
 Then the reply is only executed when the instantiate is successful with the instantiate response data. [https://github.com/terraswap/terraswap/blob/7cf47f5e811fe0c4643a7cd09500702c1e7f3a6b/contracts/terraswap\_factory/src/contract.rs#L148-L170](https://github.com/terraswap/terraswap/blob/7cf47f5e811fe0c4643a7cd09500702c1e7f3a6b/contracts/terraswap\_factory/src/contract.rs#L148-L170).
 
+Note differences from standard CosmWasm:
+* Obtaining contract data instantiated w/ sub-messages: 
+    * `msg.result.events` from `msg: Reply` passed into `#entry_point pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> StdResult<Response>` should be used to obtain data, such as `contract_address`, after instantiating a contract. Standard CosmWasm will return `MsgInstantiateContractResponse` data as protobuf binary inside `msg.result.data`. However `msg.result.data` currently returns no binary data on Secret Network v1.4. Thus, utilies such as [`parse_instantiate_response_data`](https://github.com/scrtlabs/cw-plus/blob/14a9db7e8b93039fa856cbe126ffdb230a00734a/packages/utils/src/parse_reply.rs#L106) from `cw-utils` will not work.
+
```

Comment:
Hey, I'm not sure I understand the difference any chance you can comment with an example of how the data is serialized on other cw 1 chain compared to SCRT serialization?

Reply:
The difference is that cw 1 chains have 
`msg.result.data =  **MsgInstantiateContractResponse protobuf binary**`

for example terraswap decodes that data here:
https://github.com/terraswap/terraswap/blob/48e399eaa2e89cca4da8c7b20c9f5512a47ec36d/contracts/terraswap_factory/src/contract.rs#L231

But on secret 1.4.0: `msg.result.data` just returns null

I'm guess it's a bug, based on this comment:
[
	`// note: even if contractAddr == nil then contractAddr.String() is ok
	// \o/🤷🤷‍♂️🤷‍♀️🤦🤦‍♂️🤦‍♀️`
](https://github.com/scrtlabs/SecretNetwork/blob/dc962d53060e8bd01a119be8333fcd5a1319ca1f/x/compute/internal/keeper/msg_server.go#L64)

Reply:
```
INFO  [enclave_contract_engine::wasm::engine] Invoking reply() in wasm
INFO  [enclave_contract_engine::wasm::contract] debug_print: "AAAAAAAAA!"
INFO  [enclave_contract_engine::wasm::contract] debug_print: "msg.result.unwrap().data: !"
INFO  [enclave_contract_engine::wasm::contract] debug_print: "msg.result.unwrap().events: [Event { ty: \"instantiate\", attributes: [Attribute { key: \"code_id\", value: \"3\", encrypted: true }, Attribute { key: \"contract_address\", value: \"secret1sh36qn08g4cqg685cfzmyxqv2952q6r8vqktuh\", encrypted: true }] }, Event { ty: \"wasm\", attributes: [Attribute { key: \"contract_address\", value: \"secret1sh36qn08g4cqg685cfzmyxqv2952q6r8vqktuh\", encrypted: true }] }]!"
```

![image](https://user-images.githubusercontent.com/4157455/197410805-e761620d-bec4-45e2-89db-5708dd7dc4b1.png)

Reply:
As you can see no data is returned for `msg.result.data`
But, on other chains they get: 
`msg.result.data` = **MsgInstantiateContractResponse protobuf binary**